src/formatting/closures.rs

```diff
@@ -56,7 +56,11 @@ pub(crate) fn rewrite_closure(
         }
 
         let result = match fn_decl.output {
-            ast::FnRetTy::Default(_) if !context.inside_macro() => {
+            // Try-without-block only if not a macro and original code is not a block (#4394)
+            ast::FnRetTy::Default(_)
+                if !context.inside_macro()
+                    && !context.snippet(body.span).trim().starts_with('{') =>
```

Comment:
Note that the discussion in issue 4394 was about adding a new config option to allow users to opt into this, but we cannot change the default behavior to do this automatically (which would violate the Style Guide).

You'll want to add a net-new config option (e.g. `PreserveClosureBlockWrapping` though I'm open to bikeshedding on the name) and then include a check of that option this conditional

Reply:
Sorry I didn't notice the request to all new config issue.  I now added the option with the name suggested.  As this is the first time I am adding an option I hope it was done properly.  (The automatic tests of the new changes failed on `i686-pc-windows` failed, but as far as I understand this is because of problems in the test environment.)